core/app/models/spree/inventory_unit.rb

```diff
@@ -4,9 +4,17 @@ module Spree
   # Tracks the state of line items' fulfillment.
   #
   class InventoryUnit < Spree::Base
-    PRE_SHIPMENT_STATES = %w(backordered on_hand)
-    POST_SHIPMENT_STATES = %w(returned)
-    CANCELABLE_STATES = ['on_hand', 'backordered', 'shipped']
+    class InvalidStateChange < StandardError; end
+
+    ON_HAND = 'on_hand'
+    BACKORDERED = 'backordered'
+    RETURNED = 'returned'
+    SHIPPED = 'shipped'
+    CANCELED = 'canceled'
+
+    PRE_SHIPMENT_STATES = [BACKORDERED, ON_HAND]
+    POST_SHIPMENT_STATES = [RETURNED]
+    CANCELABLE_STATES = [ON_HAND, BACKORDERED, SHIPPED]
```

Comment:
I know this is just keeping things like before, but we should consider to make this configurable in the future. Either via a method or configuration.